include/wil/registry_basic.h

```diff
@@ -1378,114 +1365,125 @@ inline HRESULT set_value_string_nothrow(HKEY key, _In_opt_ PCWSTR value_name, _I
     return set_value_string_nothrow(key, nullptr, value_name, data);
 }
 
-// TODO: write multisz setters that don't throw.
-//inline HRESULT set_value_multisz_nothrow(HKEY key, _In_ PCWSTR value_name, const ::std::vector<::std::wstring>& data) WI_NOEXCEPT
-//{
-//    const reg_view_details::reg_view_nothrow regview{key};
-//    return regview.set_value_multisz(value_name, data);
-//}
-//
-//inline HRESULT set_value_multisz_nothrow(HKEY key, _In_ PCWSTR value_name, const ::std::list<::std::wstring>& data) WI_NOEXCEPT
-//{
-//    const reg_view_details::reg_view_nothrow regview{key};
-//    return regview.set_value_multisz(value_name, data);
-//}
+#if defined(__WIL_WINREG_STL)
+inline HRESULT set_value_multistring_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_ PCWSTR value_name, const ::std::vector<::std::wstring>& data) WI_NOEXCEPT
+{
+    const reg_view_details::reg_view_nothrow regview{ key };
+    return regview.set_value_multistring(subkey, value_name, data);
+}
+inline HRESULT set_value_multistring_nothrow(HKEY key, _In_ PCWSTR value_name, const ::std::vector<::std::wstring>& data) WI_NOEXCEPT
+{
+    return set_value_multistring_nothrow(key, nullptr, value_name, data);
+}
+
+inline HRESULT set_value_multistring_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_ PCWSTR value_name, const ::std::list<::std::wstring>& data) WI_NOEXCEPT

```

Comment:
Should we:

* Just support `std::vector` and let callers move to `std::list` themselves?
* Just support `std::list` for calls like this?
* Template this function to support any enumerable type?
* Something else?